### PR TITLE
Implement RFD001 in C

### PR DESCRIPTION
**Pull request by burgerdev**

### CODE REVIEW

File: login/login.c [hunk 1]
```diff
@@ -337,6 +323,8 @@ int login_authenticate(glome_login_config_t* config, pam_handle_t* pamh,
     return failure(EXITCODE_PANIC, error_tag, "derive-or-generate-key");
   }
 
+  // TODO: read host id type instead of concatenating it to host_id.
+  char* host_id_type = NULL;
```
Comment (vvidic) [inline login/login.c:327]: Seems like host_id_type can be defined later in the next block?

**Review by vvidic — APPROVED** [review]
Looks fine by me, just update the TODO comments.

**Comment by burgerdev** [comment]
I worked on the TODOs, but that resulted in quite some more changes to API etc. Might make sense to take another pass here. Otherwise, I'll merge this next week.

File: login/login.c [hunk 2]
```diff
@@ -328,111 +302,113 @@ int login_prompt(glome_login_config_t* config, pam_handle_t* pamh,
...
-      return failure(EXITCODE_PANIC, error_tag, "malloc-host-id");
-    }
-    if (get_machine_id(host_id, HOST_NAME_MAX + 1, error_tag) < 0) {
-      return failure(EXITCODE_PANIC, error_tag, "get-machine-id");
-    }
-  }
+  // Derive content for the GLOME Login message.
+  char* message = NULL;
+  {
+    char* host_id = NULL;
```
Comment (vvidic) [inline login/login.c:320]: I wonder if this could be a separate function?

**Review by vvidic — APPROVED** [review]

**Pull request merged**